### PR TITLE
Cr ac join list shared

### LINKED ISSUE #5: 5. Como usuario, quiero participar de una lista de compras existente, asi puedo compartirlas con otra persona
Los tokens pueden ser compartidos con otros usuarios para permitirles gestionar una lista determinada.

Unirse a una lista existente consiste de lo siguiente
* El usuario ingresa un token existente
* Guarda el token en `localStorage`
* Muestra al usuario la lista existente

AC:

* Cuando el usuaruio no tiene un token ya guardado en el 'localStorage', se muestra una formulario simple en la pantalla inicial/home que le permite al usuario ingresar un token ya existente
* Ingresar el token y clickear enviar, guarda el token en el 'localStorage', efectivamente dándole control compartido de la lista
* Al enviar, muestra un error si el token no existe
* Si se obtiene un mensaje de error, permite al usuario intentar de nuevo o crear una nueva lista

### PR DESCRIPTION
**Pull request by alxmcr**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Description

Joining an existing list consists of the following:

* User enters an existing token
* Save the token to `localStorage`
* Show the user the existing list

## Related Issue

Fix #5

## Acceptance Criteria

* When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

## Changes

- Creation `FormJoinList` component, to enter a shared token about a list and add new items to some list
- New function `existCollectionByToken`, to determine whether a collection exists or not.
- (Challenge - Web Accessibility) Add `aria-live` to error messages

## Before

image

## After

image

### CODE REVIEW

File: src/components/FormJoinList/index.jsx
Comment (Lukaina) [inline src/components/FormJoinList/index.jsx]: The purpose of the code is well understood. Adding the `aria-live` for accessibility is very valuable, I tried it with the "Screen Reader" and it worked perfectly, I even got excited, well done!

File: src/pages/Home/index.jsx
Comment (Lukaina) [inline src/pages/Home/index.jsx]: Excellent name change, more descriptive.

**Review by Lukaina — COMMENTED** [review]
@alxmcr y @CrisRuedaP, excellent work, you can see the effort and I appreciate how they have solved the problems that have been presented in the code and how they have implemented the accessibility that is so important. I would just like you to please clarify step 3. On submit, show an error if the token does not exist.
Is that when the message appears: Invalid Token. Try again!?

Congratulations!

File: src/components/FormJoinList/index.jsx
Comment (CrisRuedaP) [reply src/components/FormJoinList/index.jsx]: Vero, thanks for your comments 💜. And yes, this part refers to acceptance criteria #3, if the user enters a token that is not valid, the error message is displayed, indicating that the token is not valid and can try again.

Comment (Lukaina) [reply src/components/FormJoinList/index.jsx]: Thank you very much for answering me, Cris. I get it, excellent!

**Review by Lukaina — APPROVED** [review]

File: src/components/FormJoinList/index.jsx
Comment (alxmcr) [reply src/components/FormJoinList/index.jsx]: **_Thanks, Veronica!_** I appreciate your words. 

Well, Cristina gave you a clarification, great! Thanks, Cristina for answering! 😄 .

File: src/pages/Home/index.jsx
Comment (alxmcr) [reply src/pages/Home/index.jsx]: Yes, it's easier to understand 😄 .

**Comment by macosta08** [comment]
congratulations meets all acceptance criteria, good job.

**Review by macosta08 — APPROVED** [review]

File: src/components/FormJoinList/index.jsx
Comment (alejandronanez) [inline src/components/FormJoinList/index.jsx]: I don't think you need `id=tokenChecking` since you're not using it anywhere else (I don't see it in any CSS or JS file).

If we're not using it, then, let's ✂️ it.

File: src/components/FormJoinList/index.jsx
Comment (alejandronanez) [inline src/components/FormJoinList/index.jsx]: Same comment I made on `tokenChecking`. If we don't need this `id` then let's ✂️ it!

File: src/components/Utils/firestore.js
Comment (alejandronanez) [inline src/components/Utils/firestore.js]: I'm glad you're taking care of handling errors. 👏 

**Sidenote.** In other more complex projects, you wouldn't want to `return false` in the `catch` block (this is called error swallowing). In such projects you'd want to:

- Report the error you just got your error reporting mechanisms like [Sentry](https://sentry.io/welcome/)
- Let the `error` bubble up so you see the real error in your console. You'll always want to know the **real** error so you can fix it. Getting `false` means nothing 😅

Comment (alejandronanez) [reply src/components/Utils/firestore.js]: In a more complex project, it could look like this

```js
try {
  // your optimistic code
} catch (error) {
  reportToSentry(error);
  // because you're not returning or throwing anything, the error will bubble up.
}
```

File: src/pages/Home/index.jsx
Comment (alejandronanez) [reply src/pages/Home/index.jsx]: Nice, thanks for the change!

I'd suggest `JoinListForm` as it reads more naturally.

**Review by alejandronanez — APPROVED** [review]
☝️ I didn't QA this feature, so do not merge it before getting an approval from your mentors. ☝️

I liked the code I saw here, nice work y'all. Please, consider doing the changes I suggested in my review.

Thank you! 👍

PS. If this is ready for review, please, take update this PR's state from `Draft => Ready For Review`

File: src/components/FormJoinList/index.jsx
Comment (alxmcr) [reply src/components/FormJoinList/index.jsx]: Thanks! we removed the ID! 👍

File: src/components/FormJoinList/index.jsx
Comment (alxmcr) [reply src/components/FormJoinList/index.jsx]: Thanks! we removed it! 👍

File: src/components/Utils/firestore.js
Comment (alxmcr) [reply src/components/Utils/firestore.js]: Thanks, @alejandronanez, excellent observation 💯👀 ! 

Well, we put `console.err(error)` at least now, we will have the opportunity to see errors in the console 😄😅 🥳.

(PS. Thanks, I'll research a little about `sentry`, maybe I can learn a little about it. I'm so sorry, I didn't know about it and I can see It's very useful 😄)

**Review by indiragp — APPROVED** [review]
Great Job!

File: src/components/Utils/firestore.js
Comment (alejandronanez) [reply src/components/Utils/firestore.js]: >  I'm so sorry, I didn't know about it and I can see It's very useful

Don't be,  it's hard to stay on top of everything these days.

**Pull request merged**
